src/lib/metrics-gauge.ts

```diff
@@ -0,0 +1,50 @@
+import { createGauge, type Gauge } from './util/metrics';
+
+type RestrictedRecord<T extends string[]> = Record<T[number], string>;
+type Query<R> = () => Promise<R | undefined | null>;
+type MapResult<R> = (result: R) => {
+    count: number;
+    labels: RestrictedRecord<GaugeDefinition<R>['labelNames']>;
+};
+
+type GaugeDefinition<T> = {
+    name: string;
+    help: string;
+    labelNames: string[];
+    query: Query<T>;
+    map: MapResult<T>;
+};
+
+export class DbMetricsMonitor {
+    private tasks: (() => Promise<void>)[] = [];
+    private gauges: Record<string, Gauge<string>> = {};
+
+    constructor() {}
```

Comment:
Are we planning on doing something with the constructor? It not, can we omit it?

Reply:
I went back and forth with adding some dependencies to it (like timer) and it just remained there... I think this is just details we can fix later